kpops/api/registry.py

```diff
@@ -45,32 +52,52 @@ def __getitem__(self, component_type: str) -> type[PipelineComponent]:
             msg = f"Could not find a component of type {component_type}"
             raise ClassNotFoundError(msg) from ke
 
+    @staticmethod
+    def iter_component_modules() -> Iterator[ModuleType]:
+        import kpops.components
+
+        yield kpops.components
+        yield from _iter_namespace(kpops.components)
 
-def find_class(module_name: str, baseclass: type[T]) -> type[T]:
+
+def find_class(modules: Iterable[ModuleType], base: type[T]) -> type[T]:
     try:
-        return next(_find_classes(module_name, baseclass))
+        return next(_find_classes(modules, base=base))
     except StopIteration as e:
         raise ClassNotFoundError from e
 
 
-def _find_classes(module_name: str, baseclass: type[T]) -> Iterator[type[T]]:
+def import_module(module_name: str) -> ModuleType:
     module = importlib.import_module(module_name)
+    # TODO: remove? unnecessary now
     if module.__file__ and not module_name.startswith(KPOPS_MODULE):
         file_path = Path(module.__file__)
         try:
             rel_path = file_path.relative_to(Path.cwd())
             log.debug(f"Picked up: {rel_path}")
         except ValueError:
             log.debug(f"Picked up: {file_path}")
-    for _, _class in inspect.getmembers(module, inspect.isclass):
-        if not __filter_internal_kpops_classes(
-            _class.__module__, module_name
-        ) and issubclass(_class, baseclass):
-            yield _class
+    return module
+
+
+def _find_classes(modules: Iterable[ModuleType], base: type[T]) -> Iterator[type[T]]:
```

Comment:
Should we make a type aliase from it?
```suggestion
ClassIterator = Iterator[Type[T]]

#...
def _find_classes(modules: Iterable[ModuleType], base: type[T]) -> ClassIterator:
```

Reply:
doesn't that make it a little harder to understand the function signature because it obfuscates the inner `T`? at least in my mind it's pretty straightforward with `T` in the parameters and the return

Reply:
ah, btw it would still need to be `ClassIterator[T]`

Reply:
perhaps we can find a better name for `T` though 😄

Reply:
Yeah I think a better name for `T` would help here

Reply:
done